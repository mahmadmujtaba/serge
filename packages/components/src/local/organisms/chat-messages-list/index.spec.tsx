/* global it expect */

import React from 'react'
import renderer from 'react-test-renderer'
import ChatMessageProps from '../../molecules/chat-message/types/props'

import ChatMessagesList from './index'

it('ChatMessagesList renders correctly', () => {
  it('renders component correctly', () => {
    const icons = [
      './images/default_img/forceDefault.png'
    ]
    const colors = [
      '#F00'
    ]
    const markAllAsRead = (): void => window.alert('Callback on mark all as read')
    const messages: ChatMessageProps[] = [{
      borderColor: '#ffffff',
      message: 'lorem ipsum do lor sit amet',
      timestamp: '2020-09-18T05:41:17.349Z',
      role: 'Game Control',
      privateMessage: 'Private message',
      isUmpire: true,
      isOwner: true,
    }]
    const tree = renderer
      .create(<ChatMessagesList messages={messages} colors={colors} icons={icons} onMarkAllAsRead={markAllAsRead} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
});