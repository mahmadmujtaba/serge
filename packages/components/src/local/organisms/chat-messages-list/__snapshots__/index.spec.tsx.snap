// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatMessagesList renders correctly renders component correctly 1`] = `
<div>
  <div
    className="MuiBox-root MuiBox-root-1"
  >
    <div
      className="forces-in-channel"
    >
      <img
        alt=""
        className="force-indicator role-icon"
        src="./images/default_img/forceDefault.png"
        style={
          Object {
            "backgroundColor": "#F00",
            "borderColor": "#F00",
          }
        }
      />
      <button
        className="MuiButtonBase-root MuiButton-root makeStyles-root-2 MuiButton-text MuiButton-textSecondary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Mark all read
        </span>
      </button>
    </div>
  </div>
  <div
    className="MuiBox-root MuiBox-root-3 messages-list"
    style={
      Object {
        "height": "calc(100vh - 280px)",
      }
    }
  >
    <div
      className="MuiBox-root MuiBox-root-4"
    >
      <div
        className="MuiBox-root MuiBox-root-5"
        style={
          Object {
            "float": "left",
          }
        }
      >
        <div
          className="chat-message-wrapper 
      chat-message-wrapper__other"
          style={
            Object {
              "borderLeftColor": "#FCFBEE",
            }
          }
        >
          <div
            className="message-text"
          >
            Content of message
          </div>
          <div
            className="MuiBox-root MuiBox-root-6"
          >
            <span>
              <span
                className="icon-private"
              >
                <svg
                  className="svg-inline--fa fa-user-secret"
                />
              </span>
              <span
                className="label-private"
              >
                Private message:
              </span>
            </span>
            <div
              className="MuiBox-root MuiBox-root-7"
            >
              <div
                className="MuiChip-root makeStyles-root-8 makeStyles-root-11 MuiChip-sizeSmall makeStyles-sizeSmall-9"
                onKeyDown={[Function]}
                onKeyUp={[Function]}
              >
                <span
                  className="MuiChip-label makeStyles-label-10 makeStyles-label-12 MuiChip-labelSmall"
                >
                  Game Control
                </span>
              </div>
              <span
                className="info-body info-body__other"
              >
                05:41
              </span>
            </div>
          </div>
          <div
            className="MuiBox-root MuiBox-root-13 private"
          >
            <div
              className="paragraph"
            >
              Private weather message
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
